gr/include/gnuradio/block_work_io.h

```diff
@@ -129,5 +142,107 @@ enum class work_return_code_t {
         1, /// rather than blocking in the work function, the block will call back to the
            /// parent interface when it is ready to be called again
 };
+template <typename T>
+class io_vec_wrap
+{
+private:
+    std::vector<T> _vec;
+    std::map<std::string, T*> _map;
```

Comment:
You know this may actually not be faster than just traversing the list every time. Yes, `map<>` is `O(log N)`, but there aren't that many outputs.

It's actually very likely faster to do the naive thing and iterate through the vector and do a compare.

Oh, the name isn't part of the `block_work_input`. Then `std::vector<std::string>` could be faster, using the index here to index into vector. Worth benchmarking.

Or are we expecting 100 outputs?

Reply:
The num outputs is likely less than 10, usually 1 or 2
i'll try without the map